ext/RMagick/rmutil.c

```diff
@@ -1005,6 +1005,8 @@ void rm_set_user_artifact(Image *images, Info *info)
 {
     const char *value;
 
+    rm_ensure_result(images);
```

Comment:
I really don't think that this is the place to check this. You are now introducing a side-effect in this method call. Can we revert this change?

cc @mockdeep

Reply:
@dlemstra @Watson1978 what do `rm_ensure_result` and `rm_set_user_artifact` do?

Reply:
Make sense. It certainly has side effects.

`rm_ensure_result` raise a exception if `images` is `NULL`. However, `rm_set_user_artifact` was just skip it.